src/text/non_blocking/raw_text_reader.rs

```diff
@@ -96,33 +96,45 @@ pub(crate) enum RootParseResult<O> {
     Failure(String),
 }
 
-impl<A: AsRef<[u8]>> RawTextReader<A> {
+impl From<Vec<u8>> for RawTextReader<Vec<u8>> {
+    fn from(source: Vec<u8>) -> Self {
+        RawTextReader::new(source)
+    }
+}
+
+pub trait Expandable {
+    fn expandable(&self) -> bool;
+}
```

Comment:
Very clever! Could you add a doc comment, please?

Reply:
Absolutely, added in the next commit.